### PR TITLE
Changes from 6.5.2 added to master

### PR DESCRIPTION
**Pull request by gmponos**

### CODE REVIEW

File: src/functions.php [hunk 1]
```diff
@@ -316,17 +316,15 @@ function _current_time()
...
 
 /**
- * @param int $options
- *
- * @return UriInterface
- *
  * @internal
  */
-function _idn_uri_convert(UriInterface $uri, $options = 0)
+function _idn_uri_convert(UriInterface $uri, int $options = 0): UriInterface
```
Comment (GrahamCampbell) [inline src/functions.php]: This won't work on PHP 5.5 and 5.6.

Comment (GrahamCampbell) [reply src/functions.php]: In fact, it will cause a parse error.

Comment (GrahamCampbell) [reply src/functions.php]: Oh, I missread the target as being Guzzle 6! Never mind...

Comment (gmponos) [reply src/functions.php]: Sorry the title might be confusing... but this PR is targeting master where we support only 7.. I am migrating latest changes of v6.5 to master..

**Pull request merged**